### PR TITLE
Little cleanup.

### PR DESCRIPTION
**Pull request by jtkech**

### CODE REVIEW

**Comment by jtkech** [comment]
Then re-doing tests on backgroud tasks i needed to re-use a null check in `ListShellContextsAsync()`.

File: src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs [hunk 1]
```diff
@@ -364,7 +364,12 @@ public async Task<IEnumerable<ShellContext>> ListShellContextsAsync()
         {
             var shellContexts = new List<ShellContext>();
 
-            foreach(var shellContext in _shellContexts.Values.ToArray())
+            if (_shellContexts == null)
```
[the remark this answers is not in the document]
Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs]: Ok, and it doesn't call InitializeAsync ... maybe it should.

[the remark this answers is not in the document]
Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs]: but then it will start the sites by itself which might be bad

**Comment by jtkech** [comment]
shellcontexts

**Comment by Jetski5822** [comment]
Move line 366 to 372 in that cleanup pic

**Comment by jtkech** [comment]
Okay, thanks

**Comment by Jetski5822** [comment]
Also, put the to array to a var, then check if length is 0.

If 0, Enumerable.Empty and return

Then create list.

File: src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore/OrchardCore.Environment.Shell/ShellHost.cs]: Yes, i think it's ok like this but i will think about it.

I'm still on another issue in bg tasks but i think not related to shells.

**Comment by jtkech** [comment]
So, it would be ready but the "issue" i saw in bg tasks is the same here. Now when using **in the background** a service which inject an `IHttpContextAccessor`, from this one we don't get the default `HttpContext` created when created the new scope, what we now do in an async way.

As said, it was not the case before but maybe still possible depending on what is done between creating the scope and using `_httpContextAccessor`. Again, only related to working in the background.

E.g in `LuceneIndexingService`.

httpcontext

**Pull request merged**